src/pages/home/report/ReportActionContextMenu.js

```diff
@@ -12,109 +12,119 @@ import ReportActionPropTypes from './ReportActionPropTypes';
+
+        /**
+         * A list of all the context actions in this menu.
+         */
+        this.CONTEXT_ACTIONS = [
+            // Copy to clipboard
+            {
+                text: 'Copy to Clipboard',
+                icon: ClipboardIcon,
+                successText: 'Copied!',
+                successIcon: Checkmark,
+                shouldShow: true,
+
+                // If return value is true, we switch the `text` and `icon` on
+                // `ReportActionContextMenuItem` with `successText` and `successIcon` which will fallback to
+                // the `text` and `icon`
+                onPress: () => {
+                    const message = _.last(lodashGet(this.props.reportAction, 'message', null));
+                    const html = lodashGet(message, 'html', '');
+                    const text = lodashGet(message, 'text', '');
+                    const isAttachment = _.has(this.props.reportAction, 'isAttachment')
+                        ? this.props.reportAction.isAttachment
+                        : isReportMessageAttachment(text);
+                    if (!isAttachment) {
+                        Clipboard.setString(text);
+                    } else {
+                        Clipboard.setString(html);
+                    }
+                },
+            },
+
+            // Copy chat link
+            {
+                text: 'Copy Link',
+                icon: LinkCopy,
+                shouldShow: false,
+                onPress: () => {},
+            },
+
+            // Mark as Unread
+            {
+                text: 'Mark as Unread',
+                icon: Mail,
+                shouldShow: true,
+                onPress: props.onMarkAsUnread,
+            },
+
+            // Edit Comment
+            {
+                text: 'Edit Comment',
+                icon: Pencil,
+                shouldShow: false,
+                onPress: () => {},
+            },
+
+            // Delete Comment
+            {
+                text: 'Delete Comment',
+                icon: Trashcan,
+                shouldShow: true,
+                onPress: () => {},
+            },
+        ];
+
+        this.wrapperStyle = getReportActionContextMenuStyles(this.props.isMini);
+    }
+
+    render() {
+        return this.props.isVisible && (
+            <View style={this.wrapperStyle}>
+                {this.CONTEXT_ACTIONS.map(contextAction => contextAction.shouldShow && (
+                    <ReportActionContextMenuItem
+                        icon={contextAction.icon}
+                        text={contextAction.text}
+                        successIcon={contextAction.successIcon}
+                        successText={contextAction.successText}
+                        isMini={this.props.isMini}
+                        key={contextAction.text}
+                        onPress={() => contextAction.onPress()}
+                    />
+                ))}
+            </View>
+        );
+    }
+}
```

Comment:
Lot's of the changes here are just to avoid conflicts with https://github.com/Expensify/Expensify.cash/pull/2046, which should be merging soon

Reply:
Hm, OK, I'll have a look over there too. Maybe some of my points are better made over there.